modules/tinymce/src/plugins/table/main/ts/ui/TableDialog.ts

```diff
@@ -62,18 +63,25 @@ const applyDataToElement = (editor: Editor, tableElm: HTMLTableElement, data: Ta
   }
 
   if (shouldStyleWithCss) {
-    styles['border-width'] = Utils.addPxSuffix(data.border);
+    if (borderIsZero) {
+      attrs.border = 0;
+      styles['border-width'] = '';
+    } else {
+      styles['border-width'] = Utils.addPxSuffix(data.border);
```

Comment:
Based on the DoD I think this is what we want to do
```suggestion
      styles['border-width'] = Utils.addPxSuffix(data.border);
      attrs.border = 1;
```